Client/src/component/NewReviewForm.jsx

```diff
@@ -0,0 +1,379 @@
+};
+const OVERLAY_STYLES = {
+  position: 'fixed',
+  top: 0,
+  left: 0,
+  right: 0,
+  bottom: 0,
+  backgroundColor: 'rgba(0, 0, 0, .7)',
+  zIndex: 1000,
+};
+
+const NewReviewForm = ({ showNewReviewModal, setNewReviewModal }) => {
+  const [rating, clickStar] = useState(5);
+  const [recommend, isRecomnended] = useState(undefined);
+  const [summary, changeSummary] = useState('');
+  const [body, changeReview] = useState('');
+  const [photos, uploadPhoto] = useState(null);
+  // eslint-disable-next-line camelcase
+  const [reviewer_name, inputNickname] = useState('');
+
+
+  const characterObj = [
+    {
+      0: 'Size',
+      1: 'A size too small',
+      2: '½ a size too small',
+      3: 'Perfect',
+      4: '½ a size too big',
+      5: 'A size too wide',
+    },
+    {
+      0: 'Width',
+      1: 'Too narrow',
+      2: 'Slightly narrow',
+      3: 'Perfect',
+      4: 'Slightly wide',
+      5: 'Too wide',
+    },
+    {
+      0: 'Comfort',
+      1: 'Uncomfortable',
+      2: 'Slightly uncomfortable',
+      3: 'Ok',
+      4: 'Comfortable',
+      5: 'Perfect',
+    },
+    {
+      0: 'Quality',
+      1: 'Poor',
+      2: 'Below average',
+      3: 'What I expected',
+      4: 'Pretty great',
+      5: 'Perfect',
+    },
+    {
+      0: 'Length',
+      1: 'Runs short',
+      2: 'Runs slightly short',
+      3: 'Perfect',
+      4: 'Runs slightly long',
+      5: 'Runs long',
+    },
+    {
+      0: 'Fit',
+      1: 'Runs tight',
+      2: 'Runs slightly tight',
+      3: 'Perfect',
+      4: 'Runs slightly long',
+      5: 'Runs long',
+    },
+  ];
+
+  const renderTableHeader = () => {
+    const header = Object.keys(characterObj[0]);
+    return header.map((key) => <th key={key + 1}>{key.toUpperCase()}</th>);
+  };
+  const renderTableData = () => characterObj.map((characteristic) => (
+    // eslint-disable-next-line react/no-array-index-key
+    <tr key={characteristic[0]}>
+      <td />
+      <td>
+        <button
+          type="button"
+        >
+          {characteristic[0]}
+        </button>
+      </td>
+      <td>
+        <button
+          type="button"
+        >
+          {characteristic[1]}
+        </button>
+      </td>
+      <td>
+        <button
+          type="button"
+        >
+          {characteristic[2]}
+        </button>
+      </td>
+      <td>
+        <button
+          type="button"
+        >
+          {characteristic[3]}
+        </button>
+      </td>
+      <td>
+        <button
+          type="button"
+        >
+          {characteristic[4]}
+        </button>
+      </td>
+      <td>
+        <button
+          type="button"
+        >
+          {characteristic[5]}
+        </button>
+      </td>
+
+    </tr>
+  ));
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setNewReviewModal(false);
+  };
+  const onPhotoUpload = (event) => {
+    uploadPhoto(URL.createObjectURL(event.target.files[0]));
+  };
+
+  const starRatingTextOne = () => {
```

Comment:
I think you can put all conditional logic in one single "starRatingText" function?

Reply:
I couldn't for the life of me figure out how to get it to work properly, because I need the text to show up in a specific area on click